### PR TITLE
Cpm 597 Update the migration to set nullable pim_versioning_version.resource_id

### PR DESCRIPTION
**Pull request by amigaire**
**Description (for Contributor and Core Developer)**

TODO: delete EE branch after merguez

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidSetNotNullableUuidColumns.php [hunk 1]
```diff
@@ -87,6 +87,19 @@ public function addMissing(Context $context): bool
             }
         }
 
+        if (
+            $this->tableExists('pim_versioning_version')
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidSetNotNullableUuidColumns.php:91]: This table always exist, no need to check that

File: src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidSetNotNullableUuidColumns.php [hunk 2]
```diff
@@ -87,6 +87,19 @@ public function addMissing(Context $context): bool
...
+            && !$this->isColumnNullable('pim_versioning_version', 'resource_id')
+        ) {
+            $this->connection->executeQuery(<<<SQL
+                ALTER TABLE pim_versioning_version
+                MODIFY resource_id varchar(24) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+                ALGORITHM=INPLACE,
+                LOCK=NONE;
+                SQL
+            );
+        }
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidSetNotNullableUuidColumns.php:101]: You need also to update the `getMissingCount()` method. To add 1 if the column is not nullable.

**Review by nmarniesse — COMMENTED** [review]
Can you also add a test in MigrateToUuidCommandIntegration plz? To check the column is nullable after the migration

**Review by mmetayer — APPROVED** [review]
to rebase, then GTM

File: src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidSetNotNullableUuidColumns.php [hunk 3]
```diff
@@ -87,6 +91,20 @@ public function addMissing(Context $context): bool
...
         }
 
+        if (!$this->isColumnNullable('pim_versioning_version', 'resource_id')) {
+            $this->connection->executeQuery(<<<SQL
+                ALTER TABLE pim_versioning_version
+                MODIFY resource_id varchar(24) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+                ALGORITHM=INPLACE,
+                LOCK=NONE;
+                SQL
+            );
```
Comment (amigaire) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidSetNotNullableUuidColumns.php:101]:
```suggestion
            $this->connection->executeQuery(
                <<<SQL
                ALTER TABLE pim_versioning_version
                MODIFY resource_id varchar(24) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
                ALGORITHM=INPLACE,
                LOCK=NONE;
                SQL
            );
```

**Pull request merged**